tests/test_gpu_openmp.py

```diff
@@ -263,3 +312,71 @@ def test_iso_ac(self):
         op(time=time_range.num-1, dt=dt)
 
         assert np.isclose(norm(rec), 490.55, atol=1e-2, rtol=0)
+
+    @pytest.mark.parallel(mode=[2, 4])
+    @skipif('nodevice')
+    def test_op_apply_gpu_direct(self):
+        grid = Grid(shape=(3, 3, 3))
+
+        u = TimeFunction(name='u', grid=grid, dtype=np.int32)
+
+        op = Operator(Eq(u.forward, u + 1), opt=('advanced', {'gpu-direct': True}))
+
+        # Make sure we've indeed generated OpenMP offloading code
+        assert 'omp target' in str(op)
+
+        time_steps = 1000
+        op.apply(time_M=time_steps)
+
+        assert np.all(np.array(u.data[0, :, :, :]) == time_steps)
+
+    @pytest.mark.parallel(mode=[2, 4])
+    @skipif('nodevice')
+    def test_iso_ac_gpu_direct(self):
```

Comment:
this is 99% identical to `test_iso_ac` right? never ever duplicate code except for extremely rare situations (amount of copy is tiny, eg 1-2 lines, and paste only happens once... and even then it's debatable)

so I would do something along the lines of (just to give u the idea)

```
def iso_acoustic(self, **opt_options):
    ...
    ... Operator(..., opt=('advanced', opt_options))
    ...

@skipif('nodevice')
def test_iso_acoustic(self):
    iso_acoustic()


@pytest.mark.parallel(mode=[2, 4])
@pytest.mark.parametrize('gpu_direct', [False, True])
@skipif('nodevice')
def test_mpi_iso_acoustic(self):
    opt_options = {'gpu-direct': gpu_direct}
    iso_acoustic(**opt_options)
```

Reply:
I had to change `iso_acoustic()` by `TestOperator().iso_acoustic()` otherwise I get an unkown variable error
But still I'm getting this error:
`undefined name 'gpu_direct'` for the line `opt_options = {'gpu-direct': gpu_direct}` 
Also `TestOperator.test_iso_acoustic` fails due to an `AssertionError` in `assert np.isclose(norm(rec), 490.55, atol=1e-2, rtol=0)`

Reply:
yeah... the variable is obviously undefined. Clearly, mine was just pseudocode. Once you parametrize a test, you also have to supply the variable name to the test parameters:

```
@pytest.mark.parametrize('gpu_direct', [False, True])
test_mpi_iso_acoustic(self, gpu_direct):
    ....

Reply:
ok